projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -2478,14 +2460,16 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         this._horizontalForOfs = this.combineForOfCollections(this._dataRowList, this._summaryRowList);
         const vertScrDC = this.verticalScrollContainer.dc.instance._viewContainer.element.nativeElement;
         vertScrDC.addEventListener('scroll', (evt) => { this.scrollHandler(evt); });
+        vertScrDC.addEventListener('wheel', () => { this.wheelHandler(); });
+
         this.verticalScrollContainer.onDataChanged.pipe(takeUntil(this.destroy$)).subscribe(() => {
             this.reflow();
+            this.refreshSearch(true, true);
```

Comment:
@DiyanDimitrov We need something that can re-apply highlights and active highlight in the DOM when the data in the grid changes as a result of (but not limited to):
- filtering
- sorting
- paging/page size change
- grouping
- expand/collapse of groups
- expand collapse of rows (hierarchical grid).

I suppose some actions should rebuild the cache (like filtering, expand/collapse etc.) while some might not have to (like paging). In general the idea was to handle it all in single event instead of calling it after each specific action. 
I'll think on how to differentiate between the different actions. If you have any suggestions or if you can thing of other features that change the data in view but don't require a cache rebuild let me know.

Reply:
OK, let's leave it like that, we can work on better approach to refresh the search highlights after this PR is merged.